dev-tools/composer.json

```diff
@@ -5,6 +5,7 @@
         "jangregor/phpstan-prophecy": "^1.0.0",
         "maglnet/composer-require-checker": "^4.7.1",
         "mi-schi/phpmd-extension": "^4.3.0",
+        "pdepend/pdepend": "~2.15.0",
```

Comment:
Probably I do not understand issue from your short description.
Do we run `pdepend` directly? if not, I would not like to see it here as an explicit dependency.
I spend few minutes to understand issue on pdepend. it's bullshit we need to add this workaround - what I'm concern is not to specify the version of deep dependency, but that we won't see the reason behind this entry after some time. wonder if we can document it somewhere (composer likely don't accept json5) ?



Maybe it would help when I would see what is not

Reply:
> First of all, your "source of truth" does not work, because Mess Detector is not executed in the CI workflow at any point

that's my point. The primary goal is CI, secondary is local env. If we have local env doing testing, but CI not - there is no promise we will be passing that tool checks. As we see here - we may not, because it doesn't even run (regardless, CI or CLI). I'm OK to have CLI fixed (it is a good part!), but I simply prioritise CI more (which is a no-blocker for this PR).

For me, the only part that I wonder about is
> I wonder if we can document it somewhere (composer likely don't accept json5) ?

Reply:
At least this:
remove deep dependency declaration of `pdepend/pdepend` when it's compatible with Symfony 7

(ticket)

Reply:
-=> https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7517